### ISSUE TITLE
Remaining JSON Schema Tweaks for OAS 3.1

### ISSUE THREAD

**Issue by philsturgeon** [OAI/OpenAPI-Specification#2099]
With #1977 in the bag, the majority of the work for aligning JSON Schema and OpenAPI Schema Objects has been done for v3.1 (🥳).

Here is a list of remaining clarifications and "further discussions", split into individual issues to avoid overloading any one of them with an unmanageable amount of comments. 

Description | Owner | Issue | Status
------------|--------|------|--------
~Clarify text relating to formats~ | @philsturgeon  | https://github.com/OAI/OpenAPI-Specification/issues/2093 | Somebody wanna tell me some words to write?
Make clear the difference between OAS examples object and JSON Schema examples array | @darrelmiller  | https://github.com/OAI/OpenAPI-Specification/issues/2094 | Unknown
$schema guidance | @philsturgeon  | https://github.com/OAI/OpenAPI-Specification/issues/2017 |  I'm gonna write words
Improve explanation of Reference Object | | https://github.com/OAI/OpenAPI-Specification/issues/388 | Blocked by https://github.com/json-schema-org/json-schema-spec/issues/698
So what do we do about $ref siblings | @MikeRalphson  | https://github.com/OAI/OpenAPI-Specification/issues/2089 | #2181
~Define the scope of $id resolution~ | @darrelmiller  | https://github.com/OAI/OpenAPI-Specification/issues/2092 | Done
~Clarify Nullable properties with default vales~ | @tedepstein | https://github.com/OAI/OpenAPI-Specification/pull/2115 | Done
Clarify spec wrt readOnly and writeOnly in referenced schemas | @tedepstein  | #1622
v3.1 Meta Schema (Schema Object Only) | @handrews  | https://github.com/OAI/OpenAPI-Specification/pull/2016 | 
v3.1 Meta-schema (all of OpenAPI) | | |
Deprecate discriminator?  | @tedepstein  | https://github.com/OAI/OpenAPI-Specification/issues/2143 | Generally seems like a yes, but we need to be more confident in the OpenAPI Vocab direction first? So that a "simple code gen" vocab can be used to restrict some of these keywords _or_ put discriminator back if it ever leaves.

**Comment by handrews** [comment]
Thanks, @philsturgeon. 

@darrelmiller @webron while we sort these things out, keep in mind that I expect we'll publish a `draft-handrews-json-schema-03` (or whatever) with clarifications and bugfixes.  So we can also tweak wording and add clarifications on the JSON Schema side.

If we find something that needs it, we can also release a new meta-schema (e.g. 2020-03 or something).  It's not like many implementations have caught up to 2019-09 yet, so rolling out a minor update will just mean that they'll target 2020-NN instead of 2019-09.

**Comment by handrews** [comment]
@OAI/tsc Progress on the bugfix/clarification JSON Schema draft mentioned in the previous comment can be tracked [under the draft-08-patch1 milestone](https://github.com/json-schema-org/json-schema-spec/milestone/8).

**Comment by handrews** [comment]
@philsturgeon #2016 is only a meta-schema for the schema object.
We should have a separate issue tracking a schema (not meta-schema) for the entire OAS 3.1 document.

**Comment by darrelmiller** [comment]
Open issues have been moved to appropriate milestones.
